### ISSUE TITLE
occasional test_process failure

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7575]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)| @warner reported|
|-|-|
|Trac ID|trac#337|
|Type|defect|
|Created|2003-10-16 00:17:48Z|

<details><summary>Searchable metadata</summary>

```
trac-id__337 337
type__defect defect
reporter__warner warner
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1066263468000000 1066263468000000
changetime__1468872925131048 1468872925131048
version__ 
owner__ 
cc__spiv cc__warner
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
this one failed once in the full-2.3 builder:

ERROR: testEcho (twisted.test.test_process.ProcessTestCase)
-------------------------------------------------------------------------------
Traceback: twisted.internet.defer.AlreadyCalledError, 
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/internet/process.py:390:maybeCallProcessEnded
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/mail/alias.py:236:processEnded
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/internet/defer.py:218:callback
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/internet/defer.py:264:_startRunCallbacks
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
and I believe this intermittent failure is related:

ERROR: testAbnormalTermination (twisted.test.test_process.PosixProcessTestCase)
-------------------------------------------------------------------------------
Traceback: twisted.internet.defer.AlreadyCalledError, 
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/internet/process.py:390:maybeCallProcessEnded
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/mail/alias.py:236:processEnded
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/internet/defer.py:218:callback
/home/buildbot/BuildBot/slave/full2.3/Twisted/twisted/internet/defer.py:264:_startRunCallbacks
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
These all seem related to 342 [since it seems unrelated to the test itself -- 
it's just an artifact of getting the mail problem long after, or so it seems]
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
warner, can you have a once-over over the latest logs and close
this (and 342)? I've checked manually and there don't seem to
be these failures anymore but you may have more thorough data.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
This bug, and 342, seem gone.
Please open new issues for intermittent failures if they show
up again.
Thanks
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|GitHub commented|
|-|-|

In changeset 1671a0836ce81a55b8e6de42399cdecc68bb79e0
```
#!CommitTicketReference repository="" revision="1671a0836ce81a55b8e6de42399cdecc68bb79e0"
Merge pull request #337 from twisted/8606-rodrigc-setup

Author: rodrigc
Reviewer: markrwilliams
Fixes: #8606

Integrate setup3.py functionality into setup.py
```
